lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -0,0 +1,210 @@
+                            "type": "sym",
+                            "linkpath": "sample1"
+                        }
+                    ],
+                    "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                    "authorization": {
+                        "owner": "1",
+                        "access": "private"
+                    },
+                    "@timestamp": "2021-05-01T24:00:00"
+                }
+            },
+            {
+                "_index": "riya-pbench.v6.run-toc.2021-05",
+                "_type": "_doc",
+                "_id": "3bba25b62fhdgfajgsfdty6797ed06a",
+                "_score": 0.0,
+                "_source": {
+                    "parent": "/1-default",
+                    "directory": "/1-default/sample1",
+                    "mtime": "2021-05-01T24:00:00",
+                    "mode": "0o755",
+                    "name": "sample1",
+                    "ancestor_path_elements": [
+                        "1-default"
+                    ],
+                    "files": [
+                        {
+                            "name": "result.txt",
+                            "mtime": "2021-05-01T24:00:00",
+                            "size": 0,
+                            "mode": "0o644",
+                            "type": "reg"
+                        },
+                        {
+                            "name": "user-benchmark.cmd",
+                            "mtime": "2021-05-01T24:00:00",
+                            "size": 114,
+                            "mode": "0o755",
+                            "type": "reg"
+                        }
+                    ],
+                    "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                    "authorization": {
+                        "owner": "1",
+                        "access": "private"
+                    },
+                    "@timestamp": "2021-05-01T24:00:00"
+                }
+            }
+
+        Output:
+            {
+                "directories":
+                [
+                    "sample1"
+                ],
+                "files": [
+                    {
+                        "name": "reference-result",
+                        "mtime": "2021-05-01T24:00:00",
+                        "size": 0,
+                        "mode": "0o777",
+                        "type": "sym",
+                        "linkpath": "sample1"
+                    }
+                ]
+            }
+
+        """
+        dir_list = []
+        file_list = []
+        for toc in es_json["hits"]["hits"]:
+            if toc["_source"]["directory"] == context["parent"]:
+                # Retrieve files list if present else add an empty list.
+                file_list = toc["_source"].get("files", [])
+            elif (
+                toc["_source"]["parent"] == context["parent"]
+                and "name" in toc["_source"]
```

Comment:
We really have documents with no `"name"`, or is this just paranoia? 😬

Reply:
I'd like to know that, too...how can we confirm it?

Reply:
I think the critical question is whether this is a *routine* feature of "normal" documents, or an abnormal error. In the latter case, we can just catch the `KeyError` and abort with an internal error (something's wrong with the TOC documents for that dataset, so we don't really know what we ought to report). In the former case (expected, even if rare), something like the conditional here is probably appropriate.

I may try to find this code in the indexer to confirm a bit later...

Reply:
this seems to be the only case with the '\' value, where it is a directory as well as parent.

Reply:
> this seems to be the only case with the '' value, where it is a directory as well as parent.

OK, so we only have to worry about a missing `"name"` field on the `if` clause, where we're not using it? That's good.

Reply:
yeah :)